ctapipe/image/cleaning.py

```diff
@@ -2,16 +2,65 @@
 Image Cleaning Algorithms (identification of noisy pixels)
 """
 
-__all__ = ['tailcuts_clean', 'dilate']
+__all__ = ['tailcuts_clean', 'dilate', 'fact_image_cleaning', 'ImageCleaning']
 
 import numpy as np
 from scipy.sparse.csgraph import connected_components
-
-
-def tailcuts_clean(geom, image, picture_thresh=7, boundary_thresh=5,
-                   keep_isolated_pixels=False,
-                   min_number_picture_neighbors=0):
-
+from ctapipe.core import Component
+from traitlets import Dict, CaselessStrEnum
+
+
+class ImageCleaning(Component):
+    '''
+    A component to make image cleaning configurable using traitlets.
+    '''
+
+    key_type = CaselessStrEnum(
+        default_value='camera_id',
+        values=['camera_id', 'telescope_id'],
+        help='If the keys in settings are telescope ids or camera ids'
+    ).tag(config=True)
+    settings = Dict(
+        help='A dict mapping either camera_id or telescope_id to cleaning settings.'
+        ' This are keyword arguments for the actual cleaning function plus the'
+        ' the key algorithm which can be either "tailcuts" or "fact"'
+    ).tag(config=True)
+
+    def __call__(self, geom, image, arrival_times=None, telescope_id=None):
```

Comment:
We haven't used `__call__` for other classes in the pipeline. Should we be consistent in some way? Either changing them to use `__call__`, or this one to use some member function.

Reply:
I think we probably should have used `__call__` in other places, c.f. #1024

Reply:
This was a long conversation in the past, and we ended up keeping the explicit functions (for semantic reasons), but I think having a uniform interface is fine, then algorithms that are only functions can be passed around like algorithms that are configurable Components.

Reply:
THe nice thing about `__call__` is, it doesn't matter if it's a configurable class or a function. The call is the same.
For example we found that the `HillasReconstructor` could be a simple function, but `TemplateReconstructor` cannot.

If TemplateReconstructor implements `__call__` instead of predict, both have the same interface, e.g.

```
reconstruct_stereo = simple_stereo_hillas
# or
reconstruct_stereo = TemplateReconstruction(templates='templates.hdf5')

but:
```
reconstruct_stereo(event)
```

Reply:
agreed. And we can keep the grammar semantics as you show:
`clean_image = ImageCleaner(..)` (i.e the instance is named with a verb, while the class is a noun)